src/rust-crypto/rust-crypto.ts

```diff
@@ -206,8 +206,22 @@ export class RustCrypto implements CryptoBackend {
     }
 
     public async exportRoomKeys(): Promise<IMegolmSessionData[]> {
-        // TODO
-        return [];
+        const raw = await this.olmMachine.exportRoomKeys(() => true);
+        return JSON.parse(raw);
+    }
+
+    public async importRoomKeys(keys: IMegolmSessionData[], opts?: ImportRoomKeysOpts): Promise<void> {
+        const jsonKeys = JSON.stringify(keys);
+        await this.olmMachine.importRoomKeys(jsonKeys, (progress: BigInt, total: BigInt) => {
+            const importOpt: ImportOpts = {
+                total: Number(total),
+                successes: Number(progress),
+                stage: "load_keys",
+                failures: 0,
+            };
+            opts?.progressCallback?.(importOpt);
+        });
+        return;
     }
```

Comment:
I'm somewhat concerned that we aren't doing anything with `opts.untrusted` and `opts.source` here. Do we know what they are meant to do? Is it a problem that we aren't implementing them? Should we deprecate them? Or at least leave a TODO to do something with them later?

Reply:
Will add a todo.
Not sure what opts.source was supposed to be, but I suspect it is do determine if the key was received from the Backup, from a File, or from a gossip? It's most likely interesting for a dev tool. From the crypto point of view the key is trusted or not (a trusted key would be one received from a m.room.key).

For `opts.untrusted` as for now any key imported by this method will be considered as untrusted (messages decrypted with it will have the gray shield "The authenticity of this message is not guarenteed").

But I think we might be missing the mapping of the `from_backup` flag (that is used by the sdk to decide if that key needs to be uploaded to backup).